superduperdb/misc/configs.py

```diff
@@ -0,0 +1,41 @@
+from . import config
+from dataclasses import dataclass
+from functools import cached_property
+from pathlib import Path
+from superduperdb.misc import dicts
+from typing import Any, Dict, List, Optional, Tuple, Type, Union
+import os
+
+Self = Any
+File = Union[Path, str]
+
+ROOT = Path(__file__).parents[1]
+PREFIX = 'SUPERDUPERDB_'
+SECRET_PREFIX = 'SECRET_' + PREFIX
+FILES_NAME = 'CONFIG_FILES'
+DEFAULT_FILES = str(ROOT / 'configs.json')
+SECRET_DEFAULT_FILES = str(ROOT / '.secrets.json')
+FILE_SEP = ','
+
+
+@dataclass(frozen=True)
+class ConfigSettings:
+    cls: Type
+    default_files: str
+    prefix: str
+    environ: Optional[Dict] = None
```

Comment:
unfortunately this is not enough type-wise in terms of generics.
```suggestion
    environ: Optional[Mapping[str, str]] = None
```
plus immutable

Reply:
wilco!

Overall I have been somewhat uneven with my `Dict` typing, mainly because there is no automatic testing, but also because for many of the `dicts.py` code these are actually "generic" `Dict`s that might contain many types..